pisces/data_sets.py

```diff
@@ -321,10 +326,14 @@ def load_feature_data(self, feature: str | None, id: str | None) -> Dict[str, np
             raise ValueError(f"Feature {feature} not found in {self.name}")
     
     @classmethod
-    def find_data_sets(cls, root: str | Path) -> Dict[str, 'DataSetObject']:
+    def find_data_sets(cls, 
+                       root: str | Path,
+                       ignore_startswith: List=["."], # Ignore files starting with these strings
+                       ignore_endswith: List=[".tmp"], # Ignore files ending with these strings
+                       id_templates: Dict[str, str] | str | None=None, # The template for extracting IDs from the file names. A template per feature can be provided as a dictionary
+                       id_symbol: str="<<ID>>",
```

Comment:
See my other comment about `id_template` and `id_symbol` also varying between data sets, or only being necessary for certain ones. I'm not really sure what the best way to approach this is, though. 

Providing per-set, per-feature templates seems like it will actually require `Dict[(str, str), str]` and 2 steps:
1. A call to `find_data_sets` that discovers the sets
2. A second `parse_data_sets` function (TODO) that consumes `id_templates: Dict[(str, str), str]` and `id_symbol: str | Dict[str, str]`